### PR TITLE
SCMOD-7476: Override log4j configuration

### PR DESCRIPTION
**Pull request by kusumaghoshdastidar**
Jira: https://portal.digitalsafe.net/browse/SCMOD-7476

Dev builds: http://sou-jenkins2.hpeswlab.net/job/CAFapi/view/Developer/job/CAFapi~opensuse-elasticsearch7-image~SCMOD-7476~CI/

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -73,8 +73,40 @@
         <dockerProjectVersion>${dockerVersionSeperator}${project.version}</dockerProjectVersion>
     </properties>
 
+    <dependencies>
+        <dependency>
+            <groupId>com.github.cafapi</groupId>
+            <artifactId>caf-logging-log4j2</artifactId>
+            <version>1.0.0-SCMOD-7476-SNAPSHOT</version>
```
Comment (dermot-hardy) [inline pom.xml]: Don't forget to modify before merging.

File: src/main/docker/Dockerfile [hunk 2]
```diff
@@ -46,9 +46,6 @@ COPY --from=es7-image --chown=1000:0 /usr/share/elasticsearch/ /usr/share/elasti
 RUN ES_JAVA_OPTS="$(echo ${https_proxy} | sed -E 's/^.*:\/\/(.*):([0-9]*).*/-Dhttps.proxyHost=\1 -Dhttps.proxyPort=\2/')" \
     /usr/share/elasticsearch/bin/elasticsearch-plugin install analysis-icu
 
-# Configure logging
-COPY logging.yml /usr/share/elasticsearch/config
-
```
Comment (dermot-hardy) [inline src/main/docker/Dockerfile:51]: Presumably this means the `logging.yml` file should be being removed as well?

File: pom.xml [hunk 3]
```diff
@@ -73,8 +73,40 @@
...
+                                    <groupId>com.github.cafapi</groupId>
+                                    <artifactId>caf-logging-log4j2</artifactId>
+                                    <outputDirectory>${project.build.directory}/config-files</outputDirectory>
+                                    <includes>log4j2.*</includes>
+                                </artifactItem>
+                            </artifactItems>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
```
Comment (dermot-hardy) [inline pom.xml]: I don't understand why these are being extracted from the jar.

File: src/main/docker/Dockerfile [hunk 4]
```diff
@@ -59,6 +56,12 @@ RUN chgrp 0 /usr/local/bin/docker-entrypoint.sh && \
 COPY scripts/ /opt/elasticsearch/scripts/
 RUN chmod +x /opt/elasticsearch/scripts/*
 
+# Configure logging
+COPY /maven/caf-*.jar /usr/share/elasticsearch/lib/
+COPY /maven/commons-*.jar /usr/share/elasticsearch/lib/
+COPY /maven/util-*.jar /usr/share/elasticsearch/lib/
```
Comment (dermot-hardy) [inline src/main/docker/Dockerfile]: What are we trying to exclude here?  Why isn't it just all dependencies?  If an exclusion is required would it not be better to do it in the pom file using the normal excludes construct?

File: src/main/docker/Dockerfile [hunk 5]
```diff
@@ -59,6 +56,12 @@ RUN chgrp 0 /usr/local/bin/docker-entrypoint.sh && \
...
+COPY /maven/log4j2.properties /usr/share/elasticsearch/config
```
Comment (dermot-hardy) [inline src/main/docker/Dockerfile]: I don't understand why we're not just using the configuration from the jar.

**Review by dermot-hardy — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 3 above)
Comment (kusumaghoshdastidar) [reply pom.xml]: Included log4j.properties with image instead of extracting from jar

File: src/main/docker/Dockerfile (hunk 4 above)
Comment (kusumaghoshdastidar) [reply src/main/docker/Dockerfile]: Excluded the jackson-core artifact in dependencies

File: src/main/docker/Dockerfile (hunk 5 above)
Comment (kusumaghoshdastidar) [reply src/main/docker/Dockerfile]: Elasticsearch is looking for the log4j.properties file to be in the ES 'config' folder

File: src/main/docker/Dockerfile (hunk 2 above)
Comment (kusumaghoshdastidar) [reply src/main/docker/Dockerfile:51]: Removed logging.yml

File: pom.xml [hunk 6]
```diff
@@ -115,6 +129,24 @@
...
+                                                <useTransitiveDependencies>true</useTransitiveDependencies>
+                                                <useProjectArtifact>false</useProjectArtifact>
+                                            </dependencySet>
+                                        </dependencySets>
+                                        <fileSets>
+                                            <fileSet>
+                                                <directory>${project.build.directory}/config-files</directory>
+                                                <outputDirectory>./</outputDirectory>
+                                            </fileSet>
+                                        </fileSets>
```
Comment (dermot-hardy) [inline pom.xml]: I don't think this file set is still necessary.

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**